### PR TITLE
Add help functionality 

### PR DESCRIPTION
**Pull request by baselbj**
Hi,
This will add  (help feature) to each command where php ci.php help [command] will display:
1) description
2) usage
3) Arguments, if exists
4) options, if exists

Also updated all existing commands to display proper help.

### CODE REVIEW

File: system/CLI/BaseCommand.php [hunk 1]
```diff
@@ -1,150 +1,232 @@
...
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+* THE SOFTWARE.
+*
+* @package	CodeIgniter
+* @author	CodeIgniter Dev Team
+* @copyright	Copyright (c) 2014 - 2017, British Columbia Institute of Technology (http://bcit.ca/)
```
Comment (lonnieezell) [inline system/CLI/BaseCommand.php]: Looks like your editor was removing leading spaces in docblock comments and it shouldn't have been. Please fix formatting throughout.

File: system/CLI/BaseCommand.php [hunk 2]
```diff
@@ -1,150 +1,232 @@
...
+*/
 abstract class BaseCommand
 {
+    /**
+    * The group the command is lumped under
+    * when listing commands.
+    *
+    * @var string
+    */
+    protected $group;
```
Comment (lonnieezell) [inline system/CLI/BaseCommand.php]: yup. Need to update your IDE settings. Our styleguide asks for tabs for indentation, and spaces for alignment.

File: system/CLI/BaseCommand.php [hunk 3]
```diff
@@ -1,150 +1,232 @@
...
+    * show Help include (usage,arguments,description,options)
+    *
+    *
+    * @return mixed
+    */
+    public function showHelp()
+    {
+		$spaces= "   ";
+        CLI::write(lang('CLI.helpDescription'), 'yellow');
+        CLI::write($spaces . $this->description);
```
Comment (lonnieezell) [inline system/CLI/BaseCommand.php]: Formatting in this area is all wonky, making it difficult to read. Please fix.

File: system/CLI/BaseCommand.php [hunk 4]
```diff
@@ -1,150 +1,232 @@
...
+
+		if (!empty($this->arguments)){
+			CLI::write(lang('CLI.helpArguments'), 'yellow');			
+            foreach ($this->arguments as $argument => $description) {
+                CLI::write( $spaces . CLI::color(str_pad($argument, $pad),'green') . $description,'yellow');               
+            }
+             CLI::newLine();
+        }
+		
+        if (!empty($this->options)){
```
Comment (lonnieezell) [inline system/CLI/BaseCommand.php]: opening braces should be on their own line. Please reference our style guide.

File: system/Commands/Database/MigrateCurrent.php [hunk 5]
```diff
@@ -63,6 +63,30 @@ class MigrateCurrent extends BaseCommand
...
+     * @var string
+     */
+    protected $usage = 'migrate:current [options]';
+
+    /**
+     * the Command's Arguments
+     *
+     * @var array
+     */
+    protected $arguments = array();
```
Comment (lonnieezell) [inline system/Commands/Database/MigrateCurrent.php]: Please use short array syntax throughout, i.e. `[]` instead of `array()`

**Comment by baselbj** [comment]
I fixed the issues but didn't find and place in the documentation to add the help function.

anyway the CLI help will give instruction on how to use when writing php ci.php help

**Comment by lonnieezell** [comment]
The doc page to add it to would be this one, I imagine: https://github.com/bcit-ci/CodeIgniter4/blob/develop/user_guide_src/source/general/cli_commands.rst

**Comment by baselbj** [comment]
Please Check this

**Pull request merged**